modules/virt-accessing-vmi-yaml-ssh.adoc

```diff
@@ -0,0 +1,127 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-accessing-vm-consoles.adoc
+
+[id="virt-accessing-vmi-yaml-ssh_{context}"]
+= Accessing a virtual machine instance via SSH with YAML configurations
+
+You can enable a SSH connection to a VMI without the need to expose the node port number whenever a SSH connection is made. Edit the YAML file for the VMI by adding a special label for the VMI and specifying the `masquerade` binding method. Then, create a new YAML file that provides the service for SSH connections to the VMI.
+
+The following examples show the configurations for the VMI's YAML file and the service YAML file.
+
+.Prerequisites
+* Install the OpenShift CLI (`oc`).
+* You must be in the same project as the VMI.
+* The VMI you want to access must be connected to the default pod network by using the `masquerade` binding method.
+* The VMI you want to access must be running.
+
+.Procedure
+
+. Edit the YAML file for the VMI. Specify a name for exposing the service for SSH connections and the port number. Enable the `masquerade` feature for the interface.
+
++
+.Example output
+[source,yaml]
+----
+...
+kind: VirtualMachineInstance
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vmi-ssh
+  name: vmi-ssh <1>
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - masquerade: {}
+        name: testmasquerade <2>
+        ports:
+        - name: ssh
+          port: 22 <3>
+          protocol: TCP
+      rng: {}
+    machine:
+      type: ""
+    resources:
+      requests:
+        memory: 1024M
+  networks:
+  - name: testmasquerade
+    pod: {}
+...
+----
+<1> The name for the VMI that is used to expose the details for the SSH service.
+<2> The name of the `masquerade` for the interface.
+<3> The port number used for SSH connections.
+
+. Verify that the SSH service is running:
++
+[source,terminal]
+----
+oc get vmi
+----
++
+.Example output
+[source,terminal]
+----
+NAME    AGE     PHASE       IP              NODENAME
+vmi-ssh 6s      Running     10.244.196.152  node01
+----
+
+. Check the service to find out which port was acquired by the service:
++
+[source,terminal]
+----
+$ oc get svc
+----
++
+.Example output
+[source,terminal]
+----
+NAME            TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)           AGE
+svc-ssh-exp     NodePort   10.106.236.208 <none>        20022:30093/TCP   22s
+----
+In this example, the service acquired the port number 20022.
```

Comment:
This might render fine but normally you would want a `+` before the line here (after the end of the codeblock).